### PR TITLE
[AAP-8433] Support for search/match/regex

### PR DESCRIPTION
**Pull request by mkanoor**
https://issues.redhat.com/browse/AAP-8433

Support string search in conditions.

https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_tests.html#testing-strings

- match  = pattern for searching in beginning of string
- search = pattern for searching anywhere in string
- regex  = pattern for using regular expression to do search

Updated to latest version of drools

e.g.
```
condition: event.url2 is search("groups/.*/resources/.*")
```

### CODE REVIEW

**Review by Alex-Izquierdo — DISMISSED** [review]

**Comment by ttuffin** [comment]
@mkanoor are you able to add this to the docs pls?

File: docs/conditions.rst [hunk 1]
```diff
@@ -74,6 +74,18 @@ Conditions support the following operators:
      - To check if a variable is defined
    * - is not defined
      - To check if a variable is not defined
+   * - is match(pattern,ignorecase=true)
+     - To check if the pattern exists in the beginning
```
Comment (ttuffin) [inline docs/conditions.rst:78]:
```suggestion
     - To check if the pattern exists in the beginning of the string
```

**Review by ttuffin — APPROVED** [review]
Minor suggestion, otherwise lgtm.

**Comment by Alex-Izquierdo** [comment]
Is this currently implemented in the drools side and its working? Why only have tests for the AST?

**Comment by mkanoor** [comment]
@Alex-Izquierdo on the ansible-rulebook side we only generate the AST and the AST is then forwarded to the drools for evaluation, so most of the time we would see the AST code and a new drools version. This PR updates to drools_jpy 0.1.8

**Comment by mkanoor** [comment]
@Alex-Izquierdo the tests with temporal cases take upwards of 4 minutes and the CI process cancels it after 3 minutes.

**Comment by mkanoor** [comment]
Needs #313

**Comment by Alex-Izquierdo** [comment]
> @Alex-Izquierdo the tests with temporal cases take upwards of 4 minutes and the CI process cancels it after 3 minutes.

I think it must be another problem. According to the docs, the default timeout is 360 minutes: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes

**Comment by Alex-Izquierdo** [comment]
```
E       FileNotFoundError: [Errno 2] No such file or directory: 'examples/58_string_search.yml'
```

**Comment by mkanoor** [comment]
@Alex-Izquierdo @ttuffin I was missing a file I have fixed it.

**Review by Alex-Izquierdo — APPROVED** [review]
LGTM

**Pull request merged**